### PR TITLE
[In Progress] Build and Package MSBuild Utility

### PR DESCRIPTION
**Pull request by mderoy**

### CODE REVIEW

File: external/buildscripts/build.pl [hunk 1]
```diff
@@ -1509,6 +1512,20 @@
 			system("rm -rf $distdirlibmono/gac/nunit*");
 		}
 
+		if ($buildMSBuild)
+		{
+			#Our classlibrary build is missing String.format, so discard our changes before building msbuild
+			my $currentPath = $ENV{PATH};
+			$ENV{PATH} = $startPath;
```
Comment (joncham) [inline external/buildscripts/build.pl]: You use a system Mono to build then?

File: external/buildscripts/build.pl [hunk 2]
```diff
@@ -1509,6 +1512,20 @@
...
 
+		if ($buildMSBuild)
+		{
+			#Our classlibrary build is missing String.format, so discard our changes before building msbuild
+			my $currentPath = $ENV{PATH};
+			$ENV{PATH} = $startPath;
+
+			chdir("$msbuildroot") eq 1 or die ("failed to chdir : $monoroot/external/msbuild\n");
+			system("./cibuild.sh --scope Compile --target Mono --host Mono --config Release") eq 0 or die ("failed to build msbuild");
+			system("./install-mono-prefix.sh $distdir") eq 0 or die ("failed to install mono");
```
Comment (joncham) [inline external/buildscripts/build.pl]: We do we want to install MSBuild with everything else, or to it's own install folder? I am worried about polluting out Mono build folder with MSBuild output. Thoughts? For example, I plan on removing Boo/Unityscript from there too.

File: external/buildscripts/build.pl (hunk 1 above)
Comment (mderoy) [reply external/buildscripts/build.pl]: I still need to do more digging, but I suspect our ENV is being polluted by .NET standard 2.0...maybe if we build the full 4_6 profile last we'll be okay, or explicitly set that in the env

File: external/buildscripts/build.pl (hunk 2 above)
Comment (mderoy) [reply external/buildscripts/build.pl]: I figured we'd package it the same way they do, but we should be able to copy it anywhere. I think this mostly puts things in builds/monodistribution/lib/mono/msbuild/15.0/bin, but I haven't fully compared the two monodistribution folders yet

File: external/buildscripts/build_classlibs_osx.pl [hunk 3]
```diff
@@ -36,5 +36,6 @@
 	"--artifact=1",
 	"--artifactscommon=1",
 	"--buildusandboo=1",
+	"--buildMSBuild=1",
```
Comment (mrvoorhe) [inline external/buildscripts/build_classlibs_osx.pl]: Should we add this same line to build_classlibs_wsl.pl in case someone wants to build on windows?

**Pull request closed without merging**